nexus/internal/uploader/uploader_default.go

```diff
@@ -42,24 +45,42 @@ func (u *DefaultUploader) Upload(task *UploadTask) error {
 		}
 	}(file)
 
-	req, err := retryablehttp.NewRequest(
-		http.MethodPut,
-		task.Url,
-		file,
-	)
-
-	for _, header := range task.Headers {
-		parts := strings.Split(header, ":")
-		req.Header.Set(parts[0], parts[1])
+	fileWithLen, err := NewFileWithLen(file)
+	if err != nil {
+		return err
 	}
-
+	req, err := retryablehttp.NewRequest(http.MethodPut, task.Url, fileWithLen)
 	if err != nil {
 		return err
 	}
+	for _, header := range task.Headers {
+		parts := strings.Split(header, ":")
+		req.Header.Set(parts[0], parts[1])
+	}
 
 	if _, err = u.client.Do(req); err != nil {
```

Comment:
we should check the status code here and error if it is non-200. It made debugging really hard here with the current logic

Reply:
it is done in the retry logic, we could add debugging logic there. see this line: https://github.com/wandb/wandb/pull/6401/files#diff-15b114473481614878e1b9dff0145cc72afea17107f840990627eeb3d827e82fR139

Reply:
ah ops, i even clicked into that file earlier but im bad at reading code. lgtm!